### ISSUE TITLE
preview-window set to "0,hidden" will trigger the preview command even if I don't toggle-preview

### ISSUE THREAD

**Issue by leisiji** [junegunn/fzf#3149]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish

## Problem / Steps to reproduce

When I execute `fzf --preview 'touch trigger' --preview-window=0,hidden --bind ctrl-p:toggle-preview`, fzf will execute the preview command.

But when I change the preview window size to 1 will not execute the preview command until I manully toggle the preview (`fzf --preview 'touch trigger' --preview-window=1,hidden  --bind ctrl-p:toggle-preview`). 

In conclusion, fzf has different hidden window behaviors when its window size is 1 or 0.

I tested fzf-0.35.0 not having this problem.

**Comment by LangLangBart** [comment]
> I tested fzf-0.35.0 not having this problem.

Bisecting suggests: 23d8b78ce12a6d9182a3e576de96123eb4ad176d

---

I also noticed a change in behavior with this bisect commit, when you have two hotkeys to switch previews, it used to be switching to the new preview, now it closes the old preview first and you have to press the hotkey again. 

```sh
# @junegunn
# Try this command with 23d8b78 and the previous commit, note the difference in toggling the preview
fzf --bind "?:toggle-preview+change-preview:echo question mark" --bind "tab:toggle-preview+change-preview:echo tab"
```

**Comment by junegunn** [comment]
> fzf has different hidden window behaviors when its window size is 1 or 0.

fzf running preview command in the background when the size of the window is zero is an old, lesser-known documented feature. Zero is treated differently.

https://github.com/junegunn/fzf/commit/6a75e30941af50a7197ef5e96c07a8605aec9a53

However, I didn't think someone would combine `0` with `hidden`. What's the use case?

@LangLangBart The new behavior feels correct to me. `change-preview` only changes the command template of the previewer, it shouldn't affect the status of the preview window.

**Comment by leisiji** [comment]
> However, I didn't think someone would combine 0 with hidden. What's the use case?

My neovim plugin [fzf_utils](https://github.com/leisiji/fzf_utils) uses neovim's built-in floating window as the previewer. It combines '0' with 'hidden', and the previewer not presented util I press key to toggle the previewer.
